core/frontend/src/tools/MeasureTool.ts

```diff
@@ -24,11 +24,24 @@ import { PrimitiveTool } from "./PrimitiveTool";
 import { BeButtonEvent, BeModifierKeys, CoreTools, EventHandled, InputSource } from "./Tool";
 import { ToolAssistance, ToolAssistanceImage, ToolAssistanceInputMethod, ToolAssistanceInstruction, ToolAssistanceSection } from "./ToolAssistance";
 import { GraphicType } from "../common/render/GraphicType";
+import { Format, FormatterSpec } from "@itwin/core-quantity";
 
 function translateBold(key: string) {
   return `<b>${CoreTools.translate(`Measure.Labels.${key}`)}:</b> `;
 }
 
+async function getFormatterSpecByKoQAndPersistenceUnit(koq: string, persistenceUnitName: string, fallbackQuantity: QuantityType): Promise<FormatterSpec | undefined> {
+  if (!IModelApp.formatsProvider) {
+    return IModelApp.quantityFormatter.findFormatterSpecByQuantityType(fallbackQuantity);
+  }
```

Comment:
Could `IModelApp` have a default fallback formats provider, that maps given known koq name to `QuantityType` and calls quantity formatter to get the format? That would help consumers move away from quantity types towards using `IModelApp.formatsProvider`, since it would always be available.

Reply:
we could... some QuantityTypes will be converged though, Length, Cooridnate, and LengthEngineering are all mapped to AecUnits.Length, but that's a minor trade off

Reply:
I think having formats provider always available is a very important aspect. If it can be `undefined`, then consumers of the API will always have to think about fallbacks, possibly causing them to use quantity types or some other alternatives.

Reply:
I added a BasicFormatsProvider that's init by default if no formatsProvider is passed into IModelApp.startup. It contains a limited set of formats mapped to the KoQs we agreed on, using QuantityType's formatProps under the hood. It's marked internal, we don't want people to use this as a replacement for proper impls of FormatsProviders

Reply:
Gotcha, renamed the class